Lib/asyncio/base_events.py

```diff
@@ -147,6 +149,28 @@ def _ipaddr_info(host, port, family, type, proto):
     return None
 
 
+def _interleave_addrinfos(addrinfos, first_address_family_count=1):
+    """Interleave list of addrinfo tuples by family."""
+    # Group addresses by family
+    addrinfos_by_family = collections.OrderedDict()
+    for addr in addrinfos:
+        family = addr[0]
+        if family not in addrinfos_by_family:
+            addrinfos_by_family[family] = []
+        addrinfos_by_family[family].append(addr)
+    addrinfos_lists = list(addrinfos_by_family.values())
+
+    reordered = []
+    if first_address_family_count > 1:
+        reordered.extend(addrinfos_lists[0][:first_address_family_count - 1])
+        del addrinfos_lists[0][:first_address_family_count - 1]
+    reordered.extend(
```

Comment:
I removed `_roundrobin` as a standalone function and folded it into `_interleave_addrinfo`. Do we still want to move `_interleave_addrinfo` to `staggered.py`? Now that the `interleave` option works independently of `happy_eyeballs_delay`, it feels more natural to leave `_interleave_addrinfo` here.

Reply:
I think it's OK to keep it here now